.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,18 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
If you run it on every PR that will overwrite the image even before the PR is merged, I think?

Reply:
Yes, it was the suggested flow, but I am a bit skeptical as well. We can remove the PR part.